proposals/0000-a11y-cross-platform.md

```diff
@@ -0,0 +1,133 @@
+
+* accessibilityLiveRegion
+
+Several properties are missing from both platforms:
+
+* accessibilityCompoundComponent: replace accessible-- indicates to assistive technologies that all the components inside this one should be treated as a single component.
+* accessibilityRelationships: describes how this component is related to others
+* accessibilityRange: expose information about range-based components such as sliders
+
+## Motivation
+
+All accessibility properties should be available on all supported platforms. This reduces the burden on developers to understand the nuances of accessibility support on various platforms. It also allows JavaScript components to expose rich accessibility information, rather than relying on the accessibility support in native components.
+
+## Detailed design
+
+### accessibilityViewIsModal
+
+#### Android/Fire OS
+
+When a component marks itself as modal, code in the view manager will need to traverse all other components in the view tree and mark them as importantForAccessibility=no. A window content changed event with content change types set to subtree will then need to be fired on the root window to ensure that assistive technologies receive the tree updates.
+
+It may also be necessary to install a view tree observer once the modal component is active to prevent background components from appearing. We'll need to do some real-world testing to understand how realistic this scenario might be, and whether it's worth the performance cost of managing such an observer.
+
+#### iOS
+
+This property is currently supported on iOS, so no change to iOS support is needed.
+
+### accessibilityElementsHidden
+
+#### Android/Fire OS
+
+Supporting this property in Android may be as simple as marking the component as importantForAccessibility=noHideDescendants
+
+If this doesn't match the iOS behavior, we may need to build an accessibility delegate to hide the children. This will likely be tricky, since the children will still be generating events.
+
+#### iOS
+
+This property is already supported on iOS, so no iOS changes are required.
+
+### accessibilityLiveRegion
+
+This allows components to tell assistive technologies how to react to changes in their content.
+
+#### Android/Fire OS
+
+This property is already supported on Android, so no changes in Android support are required.
+
+#### iOS
+
+We believe this functionality can be implemented using accessibility notifications on iOS. For assertive live regions, the updated text of the component can be sent immediately to iOS as an announcement notification. For polite live regions, code can wait for the AnnouncementDidFinish notification before posting the updated text as an announcement notification.
+
+### accessibilityCompoundComponent
+
+We think the current implementation, which is called "accessible", should be renamed to accessibilityCompoundComponent, to more accurately represent what it does. The functionality should remain the same.
```

Comment:
The behaviour of `accessible` has always felt a bit ambiguous to me, as on Android and Web it's equivalent to `focusable`. I don't think `accessibilityCompoundComponent` is any clearer a name than `accessible`

Reply:
> Maybe I should drop this from this proposal, and make a separate proposal covering compound component handling?

I think that sounds good. The idea of an accessibility group is something that's a little complicated on web too, because you have both screen-reader accessibility and keyboard accessibility.